### PR TITLE
Public Access Live Test Failures

### PR DESCRIPTION
**Pull request by ibrahimrabab**
# Description

### CODE REVIEW

**Review by ibrandes — APPROVED** [review]

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTests.java [hunk 1]
```diff
@@ -610,10 +610,13 @@ private static Stream<Arguments> blobUploadSupplier() {
...
 
+    // Live mode does not support copy from URL with immutability policy due to public access
+    // Revisit this and remove public access once default credential is enabled
     @RequiredServiceVersion(clazz = BlobServiceVersion.class, min = "2020-10-02")
     @Test
+    @PlaybackOnly
     public void syncCopy() {
-        vlwContainer.setAccessPolicy(PublicAccessType.CONTAINER, null);
+        //vlwContainer.setAccessPolicy(PublicAccessType.CONTAINER, null);
         sleepIfRunningAgainstService(30000); // Give time for the policy to take effect
```
Comment (ibrandes) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTests.java:620]: totally forgot to write this - we can remove all of the sleep's as we aren't waiting for the access policy to set anymore

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningAsyncTests.java [hunk 2]
```diff
@@ -645,8 +646,11 @@ private static Stream<Arguments> blobUploadSupplier() {
         return Stream.of(Arguments.of(1L), Arguments.of((Long) null));
     }
 
+    // Live mode does not support copy from URL with immutability policy due to public access
+    // Revisit this and remove public access once default credential is enabled
     @RequiredServiceVersion(clazz = BlobServiceVersion.class, min = "2020-10-02")
     @Test
+    @PlaybackOnly()
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningAsyncTests.java:653]: This test does not need to be made playback only, change like 665 to vlwBlob.GenerateSasUri() or whatever the method is in Java to generate a read-only SAS the expires in 1 hour, and this test will pass again in live mode.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTests.java [hunk 3]
```diff
@@ -610,10 +610,13 @@ private static Stream<Arguments> blobUploadSupplier() {
         return Stream.of(Arguments.of(1L), Arguments.of((Long) null));
     }
 
+    // Live mode does not support copy from URL with immutability policy due to public access
+    // Revisit this and remove public access once default credential is enabled
     @RequiredServiceVersion(clazz = BlobServiceVersion.class, min = "2020-10-02")
     @Test
+    @PlaybackOnly
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTests.java:617]: Same as above.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTests.java (hunk 1 above)
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTests.java:620]: Remove this and the previous line.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/VersioningTests.java [hunk 4]
```diff
@@ -300,7 +299,7 @@ public void beginCopyBlobsWithVersion() {
 
     @Test
     public void copyFromUrlBlobsWithVersion() {
-        blobContainerClient.setAccessPolicy(PublicAccessType.CONTAINER, null);
+        //blobContainerClient.setAccessPolicy(PublicAccessType.CONTAINER, null);
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/VersioningTests.java:302]: Delete this line.

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/DirectoryApiTests.java [hunk 5]
```diff
@@ -1680,7 +1681,9 @@ public void storageAccountAudience() {
         assertTrue(aadDirClient.exists());
     }
 
+    // need to remove Playback only once default credential is enabled
     @Test
+    @PlaybackOnly
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/DirectoryApiTests.java:1686]: Revert this change since it is not related to public acccess.

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/DirectoryAsyncApiTests.java [hunk 6]
```diff
@@ -874,7 +875,9 @@ public void storageAccountAudience() {
             .verifyComplete();
     }
 
+    // need to remove Playback only once default credential is enabled
     @Test
+    @PlaybackOnly
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/DirectoryAsyncApiTests.java:880]: Revert this change because it is not related to public access

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileApiTests.java [hunk 7]
```diff
@@ -2877,7 +2877,9 @@ public void storageAccountAudience() {
         assertTrue(aadFileClient.exists());
     }
 
+    // need to remove Playback only once default credential is enabled
     @Test
+    @PlaybackOnly
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileApiTests.java:2882]: Revert this change because it is not related to public access

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java [hunk 8]
```diff
@@ -1581,7 +1581,9 @@ public void storageAccountAudience() {
             .verifyComplete();
     }
 
+    // need to remove Playback only once default credential is enabled
     @Test
+    @PlaybackOnly
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java:1586]: Revert this change because it is not related to public access

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/ShareApiTests.java [hunk 9]
```diff
@@ -1220,7 +1220,9 @@ public void storageAccountAudience() {
         assertNotNull(infoPermission);
     }
 
+    // need to remove Playback only once default credential is enabled
     @Test
+    @PlaybackOnly
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/ShareApiTests.java:1225]: Revert this change because it is not related to public access

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/ShareAsyncApiTests.java [hunk 10]
```diff
@@ -776,7 +776,9 @@ public void storageAccountAudience() {
             .verifyComplete();
     }
 
+    // need to remove Playback only once default credential is enabled
     @Test
+    @PlaybackOnly
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/ShareAsyncApiTests.java:781]: Revert this change because it is not related to public access

**Comment by ibrahimrabab** [comment]
Overriding since the tests related to public access are passing now, other test failures are not related.
